### PR TITLE
Desktop: use evergreen build for client

### PR DESCRIPTION
**Pull request by jsnajdr**
The Desktop app has been using the fallback build of the client, and that's just a historical anomaly by now: we used to use a very old version of Electron (we regularly upgrade to the latest now) and the separate `wp-desktop` repo made build coordination difficult.

The current Electron ships a reasonably new Chrome that can run the evergreen build with no problems.

This PR:
- refactors the `req.getTarget()` function to return `evergreen` for desktop, and to make the code more logical.
- removes the legacy support for `isDesktopMonorepo` flag which is always `true` now that the `wp-desktop` repo has been retired
- gives the Desktop chunks the same filenames as web version: `[name].[hash].min.js`. There's no need to be different.
- builds the evergreen build in `yarn run build-desktop:client`

**How to test:**
Verify that the desktop app works 🙂

### CODE REVIEW

File: client/server/middleware/build-target.js
Comment (sgomes) [inline client/server/middleware/build-target.js:39]: Great work simplifying the logic here 👍

**Review by sgomes — COMMENTED** [review]
These changes look great from the multi-build point of view. Excellent work, @jsnajdr! 👍 

I'll let @nsakaimbo take a final look from the desktop app point of view, to make sure nothing unexpectedly breaks.

**Comment by nsakaimbo** [comment]
Looks great! 🎉 Thanks for the awesome work managing the transition from the separate `wp-desktop` repo.

Might want to take a look at failures in the `test-server` job before landing this, though ([link](https://app.circleci.com/pipelines/github/Automattic/wp-calypso/74595/workflows/fc8e3794-5e32-4254-bd91-d03465406a15/jobs/900082)):

```
main app Route /domain-services/:action default context sets the target in desktop mode defaults to fallback in desktop mode - main app Route /domain-services/:action default context sets the target in desktop mode defaults to fallback in desktop mode

Error: expect(received).toEqual(expected) // deep equality

Expected: "fallback"
Received: "evergreen"

// ...

```

**Review by nsakaimbo — APPROVED** [review]
LGTM, pending addressing failures in `test-server` CI. 👍

**Review by scinos — APPROVED** [review]

**Comment by jsnajdr** [comment]
Server unit tests fixed, going to merge.

**Pull request merged**
